boltdesignsystem/bolt
# DS-214 Listing Teaser Component

## Jira

https://pegadigitalit.atlassian.net/browse/DS-214

## Summary

Adds the Listing Teaser component, an interactive UI element that summarizes a particular listing.

## Details

1. The sting Teaser component only has twig templates, no JS or web components.
2. It can be used to display specific information of a particular listing, article, or thread. It is commonly used on listing pages.
3. It accommodates use cases in both Academy and Collaboration Center.

### Example usage

```twig
{% set icon_questions %}
  {% include '@bolt-components-icon/icon.twig' with {
    name: 'questions',
    size: 'xlarge',
    color: 'teal',
  } only %}
{% endset %}
{% include '@bolt-components-listing-teaser/listing-teaser.twig' with {
  signifier: icon_questions,
  headline: {
    text: 'This is a listing',
    tag: 'h3',
    size: 'xlarge',
    link_attributes: {
      href: 'https://pega.com'
    },
  },
  meta_items: [
    'Posted 8 hours 15 minutes ago',
    'Last activity: 2 minutes ago',
  ],
  status: {
    solved: true,
    locked: true,
    replies: '33',
    views: '6k',
  },
  action_menu: action_menu,  // Pass Menu component here
  share_menu: share_menu,  // Pass Share component here with display set to "menu"
  more_info: 'This is the more information. Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus',
  warning: 'Warning! This is a warning message.',
} only %}
```

## How to test

Run the branch locally, do the following:

1. Check all the docs under Listing Teaser in PL.
2. Check the twig template setup.
3. Check the schema.
4. Create a dummy page and try to use the component as instructed in the docs.
